Rozhok A. Sprint - 3

alexeyRozhok opened the pull request:
qa, oop, nio io

Commit:
IO/NIO

Commit:
QA

Commit:
OOP

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/124/

SberKotlinSchool commented on oop/src/main/kotlin/ru/sber/oop/DataClasses.kt:
```diff
@@ -2,10 +2,27 @@ package ru.sber.oop
+                this.name == castedObject?.city &&
+                this.city == castedObject?.city
+                )
+    }
```
переопределяя equals нужно и переопределять hashcode. hashcode

SberKotlinSchool commented on oop/src/main/kotlin/ru/sber/oop/DataClasses.kt:
```diff
@@ -2,10 +2,27 @@ package ru.sber.oop
+        val castedObject = other as? User
+
+        return (this.age == castedObject?.age &&
+                this.name == castedObject?.city &&
```
сравниваешь name и city

SberKotlinSchool replied:
оператор ?. достаточно только в первом сравнении применить а дальше без ? можно

SberKotlinSchool commented on oop/src/main/kotlin/ru/sber/oop/Inheritance.kt:
```diff
@@ -1,13 +1,26 @@
 //TODO: create class TownSquare here...
+class TownSquare: Room(name = "Town Square", size = 1000) {
+
+    override fun load() = "Loaded Town Square"
```
надо было запретить переопределение функции load для наследников класса TownSquare. сам класс должен быть open, а метод load быть final

SberKotlinSchool commented on oop/src/main/kotlin/ru/sber/oop/Polymorphism.kt:
```diff
@@ -1,13 +1,59 @@
+class Player(override val powerType: String,
+             override var healthPoints: Int, _name: String, _isBlessed: Boolean): Fightable {
+    val name = _name
+    val isBlessed = _isBlessed
```
эти поля лучше вынести в конструктор

alexeyRozhok replied:
в этом случае ошибка Unresolved reference: isBlessed, если параметры _isBlessed, name находятся чисто внутри конструктора находятся

SberKotlinSchool replied:
class Player(override val powerType: String,
             override var healthPoints: Int, val name: String, private val isBlessed: Boolean): Fightable {

alexeyRozhok replied:
а, затупил что-то, спасибо

Commit:
fixed hashCode(), equals()

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/129/

Commit:
fixed class Player
fixed order of modificators in Inheritance load()

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/131/

rmasgutov commented on qa/src/main/kotlin/ru/sber/qa/CertificateRequestTest.kt:
```diff
@@ -0,0 +1,54 @@
+package src.main.kotlin.ru.sber.qa
```
тесты должны быть в каталоге test, иначе попадут в сборку дистрибутива

rmasgutov commented on qa/src/main/kotlin/ru/sber/qa/CertificateRequestTest.kt:
```diff
@@ -0,0 +1,54 @@
+        val certificate = certRequestLabourBook.process(hrEmployeeNumber)
+
+        assertNotNull(certificate)
+        assertTrue(certificate.certificateRequest === certRequestLabourBook)
```
Надо бы проверить свойство data

rmasgutov commented on qa/src/main/kotlin/ru/sber/qa/CertificateRequestTest.kt:
```diff
@@ -0,0 +1,54 @@
+
+        assertNotNull(certificate)
+        assertTrue(certificate.certificateRequest === certRequestLabourBook)
+        assertTrue(certificate.processedBy == hrEmployeeNumber)
```
Равенство значений лучше писать через assertEquals

rmasgutov commented on qa/src/main/kotlin/ru/sber/qa/HrDepartmentTest.kt:
```diff
@@ -0,0 +1,103 @@
+        }
+    }
+
+    //с моками получается как то так, не понял, как очереди замокать полностью
```
очереди не требуется мокать, а вот объекты, которые там храним, можно сделать моком и проверить что он вызывается ;)

rmasgutov commented on qa/src/main/kotlin/ru/sber/qa/CertificateRequestTest.kt:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    fun processWithGetScanDataThrow() {
```
на unchecked исключению обычно не пишут тесты, если они в методе явно не создаются

Commit:
fixed QA
Правки согласно замечаниям
перенес в директорию test
в CertificateRequestTest
- убрал проверки на unchecked исключения
- добавил проверку свойства data
-HrDepartmentTest - попробовал поправить тест с processNextRequest

+ учел некоторые моменты по мокированию

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/154/
